### DISCUSSION TITLE
Pinned tabs should be bookmarks

### DISCUSSION THREAD

**Discussion by hydrielax** [zen-browser/desktop#4092] [Ideas]
# Description

Pinned tabs should be regarded as bookmarks: when I pin a tab in a workspace, the URL of the pin tab should also be visible in my bookmarks, in a special folder named "Pinned tabs/[Workspace Name]".

Essentials could also be saved directly in the root directory "Pinned tabs".

# Why?

**Conceptually, pinned tabs and bookmarks are the same thing: they are URLs I want to keep for later.** It would be easier for new users to have one unique concept rather than trying to understand the technical differences between bookmarks and pinned tabs.

Plus, merging the two concepts allows to easily bring new features for pinned tabs that already exist for bookmarks:
- rename pin tabs (#869)
- organize with folders (#3152)
- bulk-manage pin tabs though the existing bookmarks manager
- synchronize pinned-tabs of a workspace between devices (since they're bookmarks)
- not retain the last activity when exiting the tab 

This would also match Arc's native behavior of pinned tabs (#2829)

**Comment by Ilovecatz17** [comment]
I agree! I was just about to make a post just like this, but you already did it! +1

**Reply by krad246** [reply]
Glad to know other people felt similarly too.

**Reply by J-Liebl** [reply]
feel the same and was about to make a discussion about it, too. Synchronizing my pinned tabs as bookmarks on other devices would be soo nice.

**Comment by tobias-hay** [comment]
I'm going to bump this because I think this makes a lot of sense. 

Especially in regards to the fact that you can have folders & sub folders for pinned tabs, which likely wont be possible with firefoxes tab group implementation.

**Comment by sillvva** [comment]
I also agree. Nested folders are significantly superior to tab groups for organization and can be easily stored as bookmarks behind the scenes. When I used Arc, I did not miss bookmarks because I had spaces and pinned tab folders. And that felt so much better than bookmarks and tab groups ever did.

**Reply by TiermoInvin** [reply]
REAL!!

**Comment by Flohhhhh** [comment]
Agreed, I desperately want to be able to rename my pinned tabs

**Reply by eblanshey** [reply]
This is already possible by double clicking the pinned tab.

**Reply by stephanhorsthemke** [reply]
I feel a bit dumb for not having tried the double click! I was searching for this. Coming from Arc I just searched through the context menu three times and then I figured that this feature does not exist.

Also a quick search said that: https://www.reddit.com/r/zen_browser/comments/1go4u8d/option_to_rename_tabs/
I came here to see if this was proposed already.

**Comment by MadL0rd** [comment]
Same
I would really like to be able to display the bookmarks folder with the workspace name in the same way that pinned tabs are currently displayed, as well as added subfolders to this panel.

**Comment by louishough** [comment]
Same, just moved from Arc and so far the only thing I miss is being able to rename pinned tabs and group them into folders...

**Reply by tobias-hay** [reply]
> Same, just moved from Arc and so far the only thing I miss is being able to rename pinned tabs and group them into folders...

Renaming was merged (#4979) so will be out in next update probably. Waiting on Firefox for folders still.

**Reply by krad246** [reply]
If it means much to you I use 'Tab ReTitle' as a substitute. It is a firefox extension. Though native support for renaming would be better.

**Reply by tobias-hay** [reply]
> If it means much to you I use 'Tab ReTitle' as a substitute. It is a firefox extension. Though native support for renaming would be better.

Native will be out next update, check my other reply for the pr #

**Reply by krad246** [reply]
Indeed, I've enjoyed what was released so far. Haven't needed the extension anymore

**Comment by Vindelgeist** [comment]
Having a folder where you can view your pinned tabs makes a lot of sense. 

But I don't think it should be in the bookmarks area. It should be a separate icon in the toolbar or sidebar.

**Reply by TiermoInvin** [reply]
They're saying that they think merging bookmarks and pinned tabs together as well as having nested folders for pinned tabs would be a good idea. They're not saying that they should be separate from bookmarks, because they will become both bookmarks and pinned tabs. It will simply be logged as a bookmark, but still be displayed as a pinned tab in the sidebar. Or did you mean something else?

Arc Browser has the same where its bookmarks are its pinned tabs.

**Reply by Vindelgeist** [reply]
No, I meant I think bookmarks and pinned tabs should stay separate.

For me pinned tabs are always the most important tabs within a workspace, but not important enough to be permanently bookmarked.

**Reply by TiermoInvin** [reply]
If you have multiple workspaces, your pinned tabs can still be the most important tabs, just logged into bookmarks if you ever need to switch browsers, or come from them. That's if they use a naming scheme on folders that automatically turn them into spaces. Which would be a cool idea

**Reply by tobias-hay** [reply]
> No, I meant I think bookmarks and pinned tabs should stay separate.
> 
> 
> 
> For me pinned tabs are always the most important tabs within a workspace, but not important enough to be permanently bookmarked.
> 
> 

Pins are permanent though, that's like their whole purpose

**Reply by Vindelgeist** [reply]
> If you have multiple workspaces, your pinned tabs can still be the most important tabs, just logged into bookmarks if you ever need to switch browsers, or come from them. That's if they use a naming scheme on folders that automatically turn them into spaces. Which would be a cool idea

That's a good reason to have pinned tabs saved as bookmarks. If any future update actually implements that though, I think there should either be a setting to turn it off, or a way to have all your pinned tabs saved to a different menu from the bookmarks toolbar, bookmarks menu, and other bookmarks.

"That's if they use a naming scheme on folders that automatically turn them into spaces. Which would be a cool idea"

Maybe we could even have a feature where you can bookmark whole spaces and they'll all get saved to a folder in your bookmarks.

**Reply by TiermoInvin** [reply]
@Vindelgeist 

>  I think there should either be a setting to turn it off, or a way to have all your pinned tabs saved to a different menu from the bookmarks toolbar, bookmarks menu, and other bookmarks.

I completely agree.

> Maybe we could even have a feature where you can bookmark whole spaces and they'll all get saved to a folder in your bookmarks.

That would be so sick!

**Reply by hydrielax** [reply]
@Vindelgeist Having different folders management systems for pin tabs and bookmarks could be great, but it kind of goes against my suggestion. My idea was not about syncing pinned tabs with bookmarks, or bringing bookmarks features to pinned tabs: it proposes to get rid of the concept of pinned tabs, to only keep the concept of bookmarks, because the two concepts are redundant in my opinion.

Let's say you're a new user, who never used a browser before: if you want to save a page for later, should you bookmark it or pin it? I am under the impression that both do the same thing, and so keeping both just fill the browser with many options and settings for both bookmarks and pin tabs, where they could be unified (I mean, just look at the contextual menu on a tab).

Ideally, I would like that all bookmarks are assigned to a "default" workspace, so any bookmarks can be found in the side panel with vertical tabs. And when I click on a bookmark, it becomes a tab, the same way a shortcut in the taskbar of a desktop environment (Windows, MacOS, Gnome...) becomes a window when I click on it. But that would be too extreme for users wanting to keep "classic" bookmarks in a separated bookmarks bar, so that's why I proposed to put workspace bookmarks in a special folder.

Sorry in advance if my explainations are not clear, I am not a native english speaker 😅

**Comment by TiermoInvin** [comment]
Adding on a cool idea, coming from other browsers, a naming scheme with a unique starting symbol/group of symbols like "=space" or "=/." where when you name a folder of bookmarks with this starting indicator, it could automatically turn your folder into a workspace.

**Comment by PinticeBTW** [comment]
yes this would be a really good thing

**Comment by SheepDomination** [comment]
If you don't want a tab no more to be pinned, now one must remove it from my bookmark.

**Comment by Sarke** [comment]
My issue is that when I click on one and then go to a different destination in that tab, the Essentials tab becomes than new thing until I click "Reset Pinned Tab" and it goes back to the original URL.  I feel like I shouldn't have to do this, since it's pinned and knows the URL I intended to put there.  In that regard I agree that it should be more like a bookmark.

My suggestion would be that if I click through to a different site or enter a new URL without thinking, I would like it to become a normal tab and the pinned Essential remain as it's original.  Basically just demote the tab in the UI, and reset the pinned one.

**Reply by vraravam** [reply]
That's completely opposite of how I use pinned tabs. I prefer the pinned tab to remain in place, even if I navigate from the same tab to a different url. I have setup a keyboard shortcut to refresh to the pinned url `Cmd+Opt+R` so that, when I want, I can quickly go back to the pinned url in the same tab.

**Reply by Sarke** [reply]
So then you want the topic "Pinned tabs should **not** be bookmarks".

**Reply by vraravam** [reply]
I think bookmarks and pinned tabs should be kept separated - as far as the user's experience is concerned.
If the bookmarks' syncing mechanism is piggy-backed for the purpose of backing and syncing, then that's fine - as long as it remains internal and doesn't leak into the UX.
I was opining about my expectation of how I use Zen's pinned tabs (as they are right now) - which was different from how your suggestion was in your 2nd paragraph of this comment thread.

**Comment by TiermoInvin** [comment]
Damn, I never really had a problem with it changing, as when you close it,
it also resets the tab to the original pinned URL. But the idea of it sort
of acting as a bookmark in that way and opening up a new tab for any other
website besides what is pinned makes sense.

**Reply by SheepDomination** [reply]
Taking a pinned tab and making it a bookmark ?

**Comment by BigBrainAFK** [comment]
I'd also like to add that exporting and import bookmarks between different browser is easier than pinned tabs themselves. So even if I move to a browser than doesnt use bookmarks/pinned tabs as a mixed concept I still keep all the stuff I had "pinned" as bookmarks for the next browser and vice versa.

Its something I appreciated when moving to Arc. Away from it will be a pain again prob so I'd love it to be nicer in Zen.

**Comment by dkapo88** [comment]
I support the notion that pinned tabs and folders should be mirrored in bookmarks. Each workspace should be its own folder, and within that folder, there should be pin folders and links. What I disliked about Arc was the lack of synchronisation between the sidebar and my bookmark system, which prevented me from using an app like Floccus to sync my bookmarks and tabs across devices and browsers.

**Comment by anselme16** [comment]
That would also solve the ambiguous state of pinned tabs, you never know when you middle click them if they are gonna unload or disappear forever (except by the very suble hover-only symbol on the right of it).

If you took time to sort it in a folder, rename it, put the right base url inside... Deleting a pinned tab should never be a one-click action. They are clearly playing the role bookmarks were designed for.

Also, for a minimalistic browser, Zen already has too many concepts : workspaces, containers, essentials, ... replacing the pinned tabs by bookmark shortcuts would simplify and ease the understanding.

**Comment by SheepDomination** [comment]
Zen is too slow on their development, are they picking their noses; just try the features and lets get going.

**Reply by tobias-hay** [reply]
> Zen is too slow on their development, are they picking their noses; just try the features and lets get going. 

Dude it's a like a single dev working on zen, but if speed is a problem zen is open source, so start sending PRs.

**Comment by mr-cheffy** [comment]
Tabs being bookmarks works theoretically but not in real life practices, new window sync PR that got merged solves all these problems without any hacky solutions related to bookmarks.

**Reply by SheepDomination** [reply]
How does the new PR that got merged solve this issue. So it sounds as if you're saying; one can create a folder although they are always pinned ? 

And so how does Zen know that I have a bookmarks folder called "test" for example and sync the folder in the tab list which is named "test" with all my bookmarks using the new window sync ?

**Reply by mr-cheffy** [reply]
> So it sounds as if you're saying; one can create a folder although they are always pinned ?

That's exactly what folders are, yes

**Reply by SheepDomination** [reply]
You missed the other question.

**Reply by mr-cheffy** [reply]
> And so how does Zen know that I have a bookmarks folder called "test" for example and sync the folder in the tab list which is named "test" with all my bookmarks using the new window sync ?

Bookmarks don't use window sync, but zen folders goes by ID matching. You can see about it here https://github.com/zen-browser/desktop/pull/10034

**Reply by SheepDomination** [reply]
Could you not use the ID of a bookmark folder and match it up to a folder in the tab folder list ?

**Reply by mr-cheffy** [reply]
Why would we do that if the current implementation works just as good, if not, more. Tabs and bookmarks have basically no relation, how would it even work behind the scenes? how would you attach a webview to a bookmark?

**Reply by SheepDomination** [reply]
> Why would we do that if the current implementation works just as good, if not, more. Tabs and bookmarks have basically no relation, how would it even work behind the scenes? how would you attach a webview to a bookmark?

I see that Arc, never really did anything with bookmarks besides basically hide them as there focus was on "spaces" as seen in Zen and tabs. 

Arc thinking, unless that is the same for Zen is; bookmarks are not what makes using a browser good, it's how you manage your tabs. And although that is true we store bookmarks for recovery of things we occasionally visit.

**Reply by mr-cheffy** [reply]
Arc doesn't have bookmarks

**Reply by SheepDomination** [reply]
> Arc doesn't have bookmarks

That's what I was meaning, Zen direction with bookmarks is the same as Arc.

**Comment by whoamitty** [comment]
Also that would make pins, and essentials exportable by default !!!!!!!!!!!!!!!!!!!

It's a feature that to me seems so obvious to develop 😭

**Reply by vraravam** [reply]
It will mean that the exported bookmarks html file (or whatever format it contains) now will not be directly importable into another browser. This is in effect polluting the bookmarks feature - which some of us don't want. (just stating my opinion/preference so as to provide the other side of thinking - each one can have their own preference as always)

**Reply by dkapo88** [reply]
I don't know why you say this; the export and import would just be an HTML file, which is a typical bookmark file export and import.

**Reply by SheepDomination** [reply]
This which I assume is coming soon to Zen, will allow all pinned tabs to be backed up as a bookmark folder with a name. Why can't we have folders with bookmarks ?

**Reply by mr-cheffy** [reply]
New version will have a zen-sessions.jsonlz4 which works the same way almost

**Reply by SheepDomination** [reply]
Therefore Zen should restore all sessions, including pinned tabs and essentials upon a hard crash or upgrade dump (all tabs are throw out when upgrading zen). With this new JSONLZ4 file.

**Reply by mr-cheffy** [reply]
Yep, it does exactly that now

**Reply by SheepDomination** [reply]
It's a hidden Session Backup but what if the session becomes corrupt, you just restore another JSONLZ4 file ?

**Reply by mr-cheffy** [reply]
yes

**Comment by jviall** [comment]
What was the resolution to this discussion? Marked as resolved but there's no definitive answer?

**Comment by evlic** [comment]
It's a real shame that pin tabs can't sync. Using bookmarks is a viable option when there's no way to bypass Firefox's data limitations. Has the maintenance team tried this?
